### PR TITLE
Relay: Refactor QueryRenderer to use props from kiwi package

### PR DESCRIPTION
**Pull request by tbergq**
Replace footer with errorFooter.
Reconnect close modal components where they belong in component tree.

### CODE REVIEW

File: app/hotels/src/map/singleHotel/SingleMap.js [hunk 1]
```diff
@@ -5,17 +5,26 @@ import { View } from 'react-native';
...
 import type { SingleMap_hotel as HotelType } from './__generated__/SingleMap_hotel.graphql';
+import CloseModal from '../../components/CloseModal';
 
 type Props = {|
   +hotel: ?HotelType,
+  +navigation: NavigationType,
 |};
 
 const SingleMap = (props: Props) => {
+  function goBack() {
```
Comment (RobinCsl) [inline app/hotels/src/map/singleHotel/SingleMap.js]: As far as I understand, this is similar as class construction. Will that be recreated on every render?

Comment (tbergq) [reply app/hotels/src/map/singleHotel/SingleMap.js]: It probably will. But the future is with react-hooks, and hooks will be available from rn-0.59. This is how we have to create functions with hooks.

File: packages/relay/src/__tests__/__snapshots__/QueryRenderer.test.js.snap [hunk 2]
```diff
@@ -2,8 +2,13 @@
 
 exports[`QueryRenderer renders footerComponent if passed 1`] = `
```
Comment (RobinCsl) [inline packages/relay/src/__tests__/__snapshots__/QueryRenderer.test.js.snap]: Message does not seem to agree with the output?

Comment (tbergq) [reply packages/relay/src/__tests__/__snapshots__/QueryRenderer.test.js.snap]: Thanks, I will update the test 😄

Comment (tbergq) [reply packages/relay/src/__tests__/__snapshots__/QueryRenderer.test.js.snap]: 🚀

**Pull request merged**